src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public abstract class Figure implements FigureArea, FigureData {
+    protected String color;
+    protected double area;
+    protected String data;
+    protected Random random = new Random();
+
+    public String getColor() {
+        return new ColorSupplier().getRandomColor();
+    }
```

Comment:
In class `Figure` you should have only one field - `color`
you can also add constructor and getter/setter but it should not have any additional logic